core/flow/rule_manager.go

```diff
@@ -186,32 +186,38 @@ func onRuleUpdate(rules []*Rule) (err error) {
 		}
 	}()
 
-	resRulesMap := make(map[string][]*Rule, len(rules))
-	for _, rule := range rules {
-		if err := IsValidRule(rule); err != nil {
-			logging.Warn("[Flow onRuleUpdate] Ignoring invalid flow rule", "rule", rule, "reason", err.Error())
+	// ignore invalid rules
+	validResRulesMap := make(map[string][]*Rule, len(rawResRulesMap))
+	for res, rules := range rawResRulesMap {
+		if len(res) == 0 {
+			logging.Warn("[Flow onRuleUpdate] Ignoring empty resource", "rules", rules)
```

Comment:
IsValidRule() check empty Resource, Maybe this check is redundant?
```golang
func IsValidRule(rule *Rule) error {
	if rule == nil {
		return errors.New("nil Rule")
	}
	if rule.Resource == "" {
		return errors.New("empty Resource")
	}
```

Reply:
Yes, I will remove here.